tests/rustworkx_tests/graph/test_bipartite.py

```diff
@@ -0,0 +1,55 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import rustworkx
+
+
+class TestBipartite(unittest.TestCase):
```

Comment:
Can you add test cases using `rustworkx.generators.generalized_petersen_graph` citing Lemma 3 of https://arxiv.org/pdf/1008.3208.pdf (it's a well known result)? I had in mind:
* Test cases with subcases looping over n, k confirming that `G(2n, 2k + 1)` are bipartite
* Test cases with subcases looping over n, k confirming that `G(2n, 2k)`, `G(2n + 1, 2k)`, `G(2n + 1, 2k + 1)`  are not bipartite

Reply:
Done in: https://github.com/Qiskit/rustworkx/pull/1002/commits/0b94c19b68cde43a509394b955046950bd98d9ea